packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -2490,6 +2497,16 @@ export class ProjectService extends BaseService {
             throw new ForbiddenError();
         }
 
+        const cachekey = `${projectUuid}-${table}-${initialFieldId}-${search}-${JSON.stringify(
+            filters,
+        )}`;
+        if (
+            !forceRefresh &&
+            cachedUniqueSearchResults[cachekey] !== undefined
+        ) {
+            return cachedUniqueSearchResults[cachekey];
```

Comment:
refresh cache idea: 
if refreshedAt is too old, we could return whatever is in the cache, and refresh the cache in the background, for the next time this is called.